geode-core/src/main/java/org/apache/geode/internal/tcp/Connection.java

```diff
@@ -3027,13 +3027,17 @@ private void readMessage(ByteBuffer peerDataBuffer) {
         failureMsg = "ClassNotFound deserializing message";
         failureEx = ex;
         rpId = md.getRPid();
-        logger.fatal("ClassNotFound deserializing message: {}", ex.toString());
+        // log at info level first in case fatal-level alert notification becomes blocked
+        logger.info(failureMsg, failureEx);
+        logger.fatal(failureMsg, failureEx.toString());
```

Comment:
why is `toString()` called explicitly in this case but not in the other two?

Reply:
Beats me why the toString is there - I didn't add it.  I do want to see the full call stack when triaging a problem like this.  Maybe it has to do with alert listeners wanting one-liners?